### PR TITLE
Refactor based modifications - MultiDim, DataCube, DataStat

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/data/table/datatable.cpp [hunk 1]
```diff
@@ -81,20 +80,16 @@ DataTableOld::DataIndex DataTableOld::addTypedColumn(
 
 		addRow(row);
 	}
-
-	return getIndex(ColumnIndex(colIndex));
 }
 
-DataTableOld::DataIndex DataTableOld::addColumn(
-    const std::string &name,
+void DataTableOld::addColumn(const std::string &name,
```
Comment (simzer) [inline src/data/table/datatable.cpp:85]: returns from a void function. same in addColumn

Comment (schaumb) [reply src/data/table/datatable.cpp:85]: that's OK, it's well-defined.

Comment (simzer) [reply src/data/table/datatable.cpp:85]: The issue is not how defined it is, but the readability of the code. I think it's not a good practice to return with void.

Comment (simzer) [reply src/data/table/datatable.cpp:85]: But since this source file will be removed soon, it's not necessary to change it.

**Review by simzer — APPROVED** [review]

**Pull request merged**